core/dbt/parser/schemas.py

```diff
@@ -1265,4 +1279,101 @@ def parse_patch(self, block: TargetBlock[UnparsedMacroUpdate], refs: ParserRef)
         macro.created_at = time.time()
         macro.meta = patch.meta
         macro.docs = patch.docs
-        macro.arguments = patch.arguments
+
+        if getattr(get_flags(), "validate_macro_args", False):
+            self._check_patch_arguments(macro, patch)
+            macro.arguments = patch.arguments if patch.arguments else macro.arguments
+        else:
+            macro.arguments = patch.arguments
```

Comment:
Nit: I believe this is logically equivalent to
```
if getattr(get_flags(), "validate_macro_args", False):
  self._check_patch_arguments(macro, patch)

if patch.arguments
  macro.arguments = patch.arguments
```
which I think is easier to follow, but eh. Also, if it's not logically equivalent I'd like to understand under which circumstances to better understand what's going on 🙂

Reply:
It is not logically equivalent because (in principle) the marco.arguments might have been set in the past before this code runs, and then overwritten. I don't have any reason to believe that was happening, but there is also no reason to take the risk of changing an existing behavior since, if they were not set, this is indeed logically equivalent.